### PR TITLE
fix: update Talos machinery to 0.12, fix secrets persistence

### PR DESCRIPTION
**Pull request by smira**
There are several important changes:

* `talosVersion` now defaults to whatever current Talos version is vs.
`0.8` as it was before; we don't expect 0.8 being used to bootstrap new
clusters
* Talos machinery bumped to latest 0.12
* secrets for the configuration are now preserved directly as
machinery's `SecretsBundle` providing forward compatibility with future
machine configuration changes: fixes persistence for cluster.id/secret
for 0.12
* migration path from "legacy" secret format to new one
* general clean ups, less dependency on "init" node type, provider
should work fine in init-less flow.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

**Comment by smira** [comment]
draft until https://github.com/talos-systems/talos/pull/4271 is merged

File: go.mod [hunk 1]
```diff
@@ -8,15 +8,14 @@ require (
 	github.com/go-logr/logr v0.1.0
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.0
-	github.com/talos-systems/crypto v0.3.1
-	github.com/talos-systems/talos/pkg/machinery v0.11.3
-	golang.org/x/sys v0.0.0-20210816074244-15123e1e1f71
+	github.com/talos-systems/crypto v0.3.2
+	github.com/talos-systems/talos/pkg/machinery v0.12.3-0.20210920195258-7e63e43eb399
```
Comment (smira) [inline go.mod:12]: 0.12.2 + secrets bundle fix, basically can be anything like 0.12.3 once we actually release that

**Review by Unix4ever — APPROVED** [review]

**Review by rsmitty — APPROVED** [review]

**Pull request merged**